x-pack/plugins/lens/public/visualizations/partition/visualization.tsx

```diff
@@ -163,18 +168,22 @@ export const getPieVisualization = ({
   triggers: [VIS_EVENT_TO_TRIGGER.filter],
 
   initialize(addNewLayer, state, mainPalette) {
-    return (
-      state || {
-        shape: PieChartTypes.DONUT,
-        layers: [
-          newLayerState(
-            addNewLayer(),
-            mainPalette?.type === 'colorMapping' ? mainPalette.value : getColorMappingDefaults()
-          ),
-        ],
-        palette: mainPalette?.type === 'legacyPalette' ? mainPalette.value : undefined,
-      }
-    );
+    return state
```

Comment:
nit: in this case I would suggest a more expressive `if` rather than a ternary.
Not strong opinion tho.

```suggestion
  if(state){
    return convertToRuntime(state);
    }
    return { ... }
```